neu-self-image-experiments/sie-firebase
# SIE-181 Implement facial detection feedback loop

This PR goes with the following [PR](https://github.com/neu-self-image-experiments/sie-backend/compare/qhoang-sie-181-facial-detection-feedback-loop?expand=1) in `sie-backend` repo to implement instant facial detection feedback.

**What's in this PR?**
- I added a function that observes the `onSnapshot` change in `facial_detection_status` attribute of a particular user doc to show instant facial detection feedback.
- Some misc changes to (1) display facial detection feedback (2) enable/disable upload button upon image requirements satisfaction.

**How to review?**
- Please look over my syntax to make sure I did not miss anything @fernandowinfield @theodore3131 
- The styling of a disabled button doesn't look really good right now. Could someone please help me with this? @alecherryy

**How to test?**
- To manually test:
  - `npm start` and go to http://localhost:3000/9iG2wajKXkjLXiLQfwgj/36qwRoyuSzcHzvCAapya4hrJBgQ2.
  - Navigate through the steps of the Wizard until step 3 is reached (photo uploading).
  - Make sure that there is no 'Upload' button and that the 'Next' button on the bottom left corner is disabled.
  - Use the webcam to take a photo that should not pass face detection. Then the 'Upload' button should appear. Click it.
  - Wait until the loading disappears and corroborate that the feedback message shows what the error was.
  - Retake or reupload a photo that should pass facial detection. Click 'Upload' again and wait.
  - Finally a success message should appear and the 'Next' button should become enabled.

**NOTE**
- Since we want to ONLY do facial detection checking here and postpone stimuli fetching to later, we might need to (1) move it somewhere else or (2) save the stimuli url in some kind of a global state? This is because per our discussion, participant will move on to pre-survey AFTER uploading their image and the image processing will just run quietly in the background. @fernandowinfield @theodore3131 @cmutuc25
